src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -112,6 +133,70 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, MSBuild.TaskLog
             return !Log.HasLoggedErrors;
         }
 
+        public async Task UploadAssets(ITaskItem item, SemaphoreSlim clientThrottle, bool allowOverwrite = false)
+        {
+            string relativeBlobPath = $"{feed.RelativePath}{item.GetMetadata("RelativeBlobPath").Replace("\\", "/")}"; 
+            if (string.IsNullOrEmpty(relativeBlobPath))
+                throw new Exception(string.Format("Metadata 'RelativeBlobPath' is missing for item '{0}'.", item.ItemSpec));
+
+            Log.LogMessage($"Uploading {relativeBlobPath}");
+
+            await clientThrottle.WaitAsync();
+
+            CloudTestTasks.AzureBlobLease blobLease = new CloudTestTasks.AzureBlobLease(feed.AccountName, feed.AccountKey, string.Empty, feed.ContainerName, relativeBlobPath, Log, "15", "5000");
+            bool blobExists = await feed.CheckIfBlobExists(relativeBlobPath);
+            bool isLeaseRequired = allowOverwrite && blobExists;
+            string leaseId = null;
+
+            if (isLeaseRequired)
+            {
+                try
+                {
+                    leaseId = blobLease.Acquire();
+                    Log.LogMessage($"Obtained lease ID {leaseId} for {relativeBlobPath}.");
+                }
+                catch (Exception)
+                {
+                    Log.LogError($"Unable to obtain lease on {relativeBlobPath}");
```

Comment:
If this method doesn't exit here, doesn't that means it continues to try to upload something while it knows that someone else has the lease?